worlds/dkc3/__init__.py

```diff
@@ -74,24 +79,25 @@ def stage_assert_generate(cls, multiworld: MultiWorld):
 
     def _get_slot_data(self):
         return {
-            #"death_link": self.world.death_link[self.player].value,
+            #"death_link": self.options.death_link.value,
             "active_levels": self.active_level_list,
         }
 
     def fill_slot_data(self) -> dict:
         slot_data = self._get_slot_data()
-        for option_name in dkc3_options:
-            option = getattr(self.multiworld, option_name)[self.player]
+        for option_name in (attr.name for attr in dataclasses.fields(DKC3Options)
+                            if attr not in dataclasses.fields(PerGameCommonOptions)):
+            option = getattr(self.options, option_name)
             slot_data[option_name] = option.value
```

Comment:
```suggestion
        slot_data.update(self.options.as_dict([attr.name for attr in dataclasses.fields(DKC3Options)
                            if attr not in dataclasses.fields(PerGameCommonOptions)])
```
Not essential to change, but better to use the function in case someone goes looking at DKC3 for reference.

Reply:
```suggestion
        slot_data.update(self.options.as_dict(
        							*[option_name
      							     for option_name in self.options_dataclass.type_hints
      							     if option_name not in PerGameCommonOptions.type_hints])``` github formatting is awful so not sure if that looks right